src/shared/condition.c

```diff
@@ -592,24 +652,40 @@ static int condition_test_firmware(Condition *c, char **env) {
                         return false;
                 } else
                         return true;
-        } else if ((dtc = startswith(c->parameter, "device-tree-compatible("))) {
-                _cleanup_free_ char *dtcarg = NULL;
+        } else if ((arg = startswith(c->parameter, "device-tree-compatible("))) {
+                _cleanup_free_ char *dtc_arg = NULL;
                 char *end;
 
-                end = strchr(dtc, ')');
+                end = strchr(arg, ')');
                 if (!end || *(end + 1) != '\0') {
-                        log_debug("Malformed Firmware condition \"%s\"", c->parameter);
+                        log_debug_errno(SYNTHETIC_ERRNO(EINVAL), FIRMWARE_MALFORMED_MSG, c->parameter);
```

Comment:
SYNTHETIC_ERRNO() is only for cases where `return log_…` is used.

Reply:
Adjusted